setting testEnvironment to node causes jest functions to be undefined

<!-- Love Jest? Please consider supporting our collective: 👉  https://opencollective.com/jest/donate -->

## 🐛 Bug Report
When setting `testEnvironment` to node and running a test I get `ReferenceError: test is not defined`. If I remove `testEnvironment: 'node'` from the config it runs fine. The same applies to `describe`, `beforeAll`...etc.

**Note**: it works fine on node 10.16.0

```
 FAIL  test/1.test.js
  ● Test suite failed to run

    ReferenceError: test is not defined

    > 1 | test('test1', () => {
        | ^
      2 |   expect(1).toBe(1);
      3 | });

      at Object.<anonymous> (test/1.test.js:1:1)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.333s
Ran all test suites.
npm ERR! Test failed.  See above for more details.
```

## To Reproduce

Steps to reproduce the behavior:

jest.config.js
```
module.exports = {
  testEnvironment: 'node'
}
```

test/1.test.js
```
test('test1', () => {
  expect(1).toBe(1);
});
```

## Expected behavior
<!-- A clear and concise description of what you expected to happen. -->
Tests should run without errors when testEnvironment is set to node.

## Link to repl or repo (highly encouraged)

<!--
Please provide either a [repl.it demo](https://repl.it/languages/jest) or a minimal repository on GitHub.
Issues without a reproduction link are likely to stall.
-->

## envinfo

<!--
Run npx envinfo --preset jest
Paste the results here:
-->

```
  System:
    OS: macOS 10.15.3
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  Binaries:
    Node: 10.14.0 - /usr/local/bin/node
    npm: 6.13.7 - /usr/local/bin/npm
  npmPackages:
    jest: ^25.1.0 => 25.1.0 
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@
 
 ### Fixes
 
+- `[jest-environment-node]` Remove `getVmContext` from Node env on older versions of Node ([#9706](https://github.com/facebook/jest/pull/9706))
+
 ### Chore & Maintenance
 
 ### Performance

--- a/packages/jest-environment-node/package.json
+++ b/packages/jest-environment-node/package.json
@@ -21,7 +21,11 @@
     "@jest/fake-timers": "^25.2.1",
     "@jest/types": "^25.2.1",
     "jest-mock": "^25.2.1",
-    "jest-util": "^25.2.1"
+    "jest-util": "^25.2.1",
+    "semver": "^6.3.0"
+  },
+  "devDependencies": {
+    "@types/semver": "^6.2.1"
   },
   "engines": {
     "node": ">= 8.3"

--- a/packages/jest-environment-node/src/index.ts
+++ b/packages/jest-environment-node/src/index.ts
@@ -14,6 +14,7 @@ import {
   LolexFakeTimers,
 } from '@jest/fake-timers';
 import type {JestEnvironment} from '@jest/environment';
+import {lt as semverLt} from 'semver';
 
 type Timer = {
   id: number;
@@ -119,6 +120,13 @@ class NodeEnvironment implements JestEnvironment {
   getVmContext(): Context | null {
     return this.context;
   }
+}
+
+// node 10 had a bug in `vm.compileFunction` that was fixed in https://github.com/nodejs/node/pull/23206.
+// Let's just pretend the env doesn't support the function.
+// Make sure engine requirement is high enough when we drop node 8 so we can remove this condition
+if (semverLt(process.version, '10.14.2')) {
+  delete NodeEnvironment.prototype.getVmContext;
 }
 
 export = NodeEnvironment;

